cylc/flow/cfgspec/globalcfg.py

```diff
@@ -1994,26 +1994,34 @@ def platform_dump(
         """Print informations about platforms currently defined.
         """
         if print_platform_names:
-            with suppress(ItemNotFoundError):
-                self.dump_platform_names(self)
+            self.dump_platform_names(self)
         if print_platforms:
-            with suppress(ItemNotFoundError):
-                self.dump_platform_details(self)
+            self.dump_platform_details(self)
 
     @staticmethod
     def dump_platform_names(cfg) -> None:
         """Print a list of defined platforms and groups.
         """
-        platforms = '\n'.join(cfg.get(['platforms']).keys())
-        platform_groups = '\n'.join(cfg.get(['platform groups']).keys())
-        print(f'{PLATFORM_REGEX_TEXT}\n\nPlatforms\n---------', file=stderr)
-        print(platforms)
-        print('\n\nPlatform Groups\n--------------', file=stderr)
-        print(platform_groups)
+        platforms, platform_groups = None, None
+        with suppress(ItemNotFoundError):
+            platforms = '\n'.join(cfg.get(['platforms']).keys())
```

Comment:
This should never happen because `localhost` is hardcoded into `cfgspec/globalconfig.py`.

Demonstrate by

```
CYLC_CONF_PATH=$HOME/intentionally_empty cylc config --platform-names
```

Reply:
Ok, I've force-pushed latest commit to simplify this